eureka-core/src/main/java/com/netflix/eureka/cluster/PeerEurekaNode.java

```diff
@@ -98,11 +131,16 @@ public PeerEurekaNode(PeerAwareInstanceRegistry registry, String name, String se
      * @throws Exception
      */
     public void register(final InstanceInfo info) throws Exception {
-        registerProcessor.process(new InstanceReplicationTask(name, Action.Register, info, null, true) {
-            public HttpResponse<Void> execute() {
-                return replicationClient.register(info);
-            }
-        });
+        long expiryTime = System.currentTimeMillis() + info.getLeaseInfo().getRenewalIntervalInSecs() * 1000;
+        batchingDispatcher.process(
+                "register#" + info.getId(),
```

Comment:
The ids here should be a concat of appName + instanceId as their combination uniquely identifies.